Add CROWDLAB for LLM Evaluation notebook (#91)
cleanlab/examples: README.md

## Table of Contents

|     | Example                                                                                        | Description                                                                                                                                                                                                                                                                  |

> | --- | ---------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

Reviewer: This header was missing for some reason, so the Table of contents was broken.  Before requesting PR reviews, please do always look at your PRs in the GH viewer under `Files Changed` to see what the delta is, and also imagine you are the Reviewer for ~10min.

Author: Ack, sorry!

> 

Reviewer: add this notebook to `ignore_folders` in:

https://github.com/cleanlab/examples/blob/master/run_all_notebooks.py

This will prevent auto-execution of this notebook. We auto-execute all other notebooks in this repo whenever we push new stable release of cleanlab package to ensure they run well.

|     | Example                                                                                        | Description                                                                                                                                                                                                                                                                  |
| --- | ---------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | 1   | datalab                              | Use Datalab to detect various types of data issues in (a subset of) the Caltech-256 image classification dataset.                        |

Reviewer: I think you removed all the links for some reason from the Table of Contents?

| 10   | [huggingface_keras_imdb](huggingface_keras_imdb/huggingface_keras_imdb.ipynb)                                             |  CleanLearning for text classification with Keras Model + pretrained BERT backbone and Tensorflow Dataset.         |
| 11   | [fasttext_amazon_reviews](fasttext_amazon_reviews/fasttext_amazon_reviews.ipynb)                         | Finding label errors in Amazon Reviews text dataset using a cleanlab-compatible [FastText model](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/models/fasttext.py).                                                                                                    |
| 12   | [multiannotator_cifar10](multiannotator_cifar10/multiannotator_cifar10.ipynb)                                             | Iteratively improve consensus labels and trained classifier from data labeled by multiple annotators.                                                            |

> | 13 | [llm_evals_w_crowdlab](llm_evals_w_crowdlab/llm_evals_w_crowdlab.ipynb) | LLM Evals with humans, AI judges, and GPT token probabilities. Evaluate an LLM from multiple human/AI reviewers of varying competency by using CROWDLAB and GPT token probabilities. |

Reviewer: ```suggestion
| 13 | [llm_evals_w_crowdlab](llm_evals_w_crowdlab/llm_evals_w_crowdlab.ipynb) | Reliable LLM Evaluation with multiple human/AI reviewers of varying competency (via CROWDLAB and LLM-as-judge GPT token probabilities). |
```

| 21  | [non_iid_detection](non_iid_detection/non_iid_detection.ipynb)  | Use Datalab to detect non-IID sampling (e.g. drift) in datasets based on numeric features or embeddings. |
| 22  | [object_detection](object_detection/README.md)  | Train Detectron2 object detection model for use with cleanlab. |
| 23  | [semantic segmentation](segmentation/training_ResNeXt50_for_Semantic_Segmentation_on_SYNTHIA.ipynb)  | Train ResNeXt semantic segmentation model for use with cleanlab. |

> | 24 | [llm_evals_w_crowdlab](llm_evals_w_crowdlab/llm_evals_w_crowdlab.ipynb) | Uses GPT4o and CROWDLAB to evaluate language models on a dataset labeled by multiple annotators. |

Reviewer: Let's make this notebook #12.
Right before the other multiannotator notebook.

The file name is fine, but the notebook itself should start with a title. I recommend same title as your corresponding article.
Then you can use this same title as the Table of Content description. Note we'll want to preserve the canonical SEO term as much as possible throughout title/writing: LLM Evaluation, LLM Evals, Evaluating LLMs.

And you don't need of the special metadata for this notebook. The special metadata is only for Tutorial notebooks that appear on doc.cleanlab.ai, this is an Example not a Tutorial.

For the Example notebooks in this repo: you should have all the outputs nicely rendered. 

For tutorial notebooks (in main cleanlab repo): you should have the outputs rendered for the initial PR just to ease review, but the final merged version should have all outputs cleared. Tutorial notebooks are auto-executed during CI to render their outputs on docs.cleanlab.ai

Reviewer: Also I'd reduce references to "dataset labeled by multiple annotators" and replace them with "multiple human/AI reviewers evaluating LLM outputs"

Author: ✔️ - moved notebook
✔️ - name change - Made _almost_ the same as Medium article but left out "cutting edge" which I think is efficacious for the medium article but not for this tutorial. 
✔️ - `Then you can use this same title as the Table of Content description`
⚠️ ❓ - `For the Example notebooks in this repo: you should have all the outputs nicely rendered`. The outputs are all rendered here, but not 100% in order as I fixed some cells. Re-running is going to be annoying bc we make a bunch of OpenAI calls so I'd respectfully ask to allow current state of "nice-ish outputs but not in order" 
✔️ -  reduce references to "dataset labeled by multiple annotators" - NO-OP given title change.

Reviewer: You don't need to re-execute anything, but do try to make the notebook outputs look better in Github viewer.
Here are some cells you can clear the outputs of:

<img width="1080" alt="Screen Shot 2024-07-01 at 8 42 54 PM" src="https://github.com/cleanlab/examples/assets/1390638/f6930fa3-1b45-4ccf-9d52-2ae774e270c1">

<img width="1009" alt="Screen Shot 2024-07-01 at 8 43 31 PM" src="https://github.com/cleanlab/examples/assets/1390638/4ee39512-571c-4abd-b239-0b484664c0c5">

This nit is mostly for future reference
